src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs

```diff
@@ -176,5 +176,41 @@ public void FractionDigitsMismatch_Throws()
             Assert.Contains("fractionDigits", ex.Message);
             Assert.DoesNotContain("totalDigits", ex.Message);
         }
+
+        [Fact]
+        public void TotalDigitsParseValue_Succeeds()
+        {
+            string schema = @"<?xml version='1.0' encoding='utf-8' ?>
+<xs:schema elementFormDefault='qualified'
+           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
+    <xs:simpleType name='foo'>
+        <xs:restriction base='xs:decimal'>
+            <xs:totalDigits value='8' fixed='true'/>
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:simpleType name='bar'>
+        <xs:restriction base='foo'>
+            <xs:totalDigits value='8'/>
+        </xs:restriction>
+    </xs:simpleType>
+</xs:schema>
+";
+            XmlSchemaSet ss = new XmlSchemaSet();
+            ss.Add(null, XmlReader.Create(new StringReader(schema)));
```

Comment:
was the test failing before the fix?

Reply:
@krwq, yes.

It was throwing an InvalidCastException.  After updating the test per feedback from @buyaa-n above, I "undid" the commit that fixed this InvalidCastException, confirmed the test failed, discarded the "undo" of the fix, and confirmed that the test now passes.

It is commit 09150e84917 which fixed the InvalidCastException:

> diff --git a/src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs b/src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs
> index e04a2f1a7f2..050a82464d9 100644
> --- a/src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs
> +++ b/src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs
> @@ -309,7 +309,7 @@ internal void CompileTotalDigitsFacet(XmlSchemaFacet facet)
> 
>                  if ((_baseFixedFlags & RestrictionFlags.TotalDigits) != 0)
>                  {
> -                    if (!_datatype.IsEqual(_datatype.Restriction.TotalDigits, _derivedRestriction.TotalDigits))
> +                    if (_datatype.Restriction.TotalDigits != _derivedRestriction.TotalDigits)
>                      {
>                          throw new XmlSchemaException(SR.Sch_FacetBaseFixed, facet);
>                      }
>